rust/worker/src/types/segment.rs

```diff
@@ -78,14 +81,19 @@ impl TryFrom<chroma_proto::Segment> for Segment {
                 return Err(SegmentConversionError::InvalidUuid);
             }
         };
-
+        let mut file_paths = HashMap::new();
+        for (key, value) in proto_segment.file_paths {
```

Comment:
you should be able to use drain() so you can avoid cloning

Reply:
proto_segment is not mutable, I don't think we can drain()

Reply:
you own the object in this contet, you can rebind mutable. 

let mut proto_segment = proto_segment.